include/class.dispatcher.php

```diff
@@ -27,6 +27,7 @@ function __construct($file=false) {
     }
 
     function resolve($url, $args=null) {
+        $url = urldecode($url); // Decode URL for accurate matching
```

Comment:
I would move this to right above foreach - line 37.  Simply to group code blocks that are used together.